.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on: [push, pull_request]
```

Comment:
No need to test it on every push, only push to main branch and pull_request

Reply:
```yaml
name: CI

on:
  push:
    branches:
      - main   
  pull_request:  
```
will this resolve the issue, this changes will test push only to the main branch and all the pull request from any branch
I'm little confused if you can help me clarify it would be helpful